addons/web_editor/static/src/js/widgets/widgets.js

```diff
@@ -1520,6 +1521,35 @@ var LinkDialog = Dialog.extend({
             class: data.classes.replace(/float-\w+/, '') + ' o_btn_preview',
         }).html((data.label && data.label.length) ? data.label : data.url);
     },
+    /**
+     * @private
+     */
+    _adaptPageAnchor: function () {
+        var self = this;
+        var urlInputValue = this.$('#o_link_dialog_url_input').val();
+        var $pageAnchor = this.$('.o_link_dialog_page_anchor');
+        var isFromWebsite = _.str.startsWith(urlInputValue, '/');
+        var anchorsArray = ['', '#top', '#bottom'];
+        var $selectMenu = this.$('select[name="link_anchor"]');
+
+        $pageAnchor.toggleClass('d-none', !isFromWebsite);
+        if (isFromWebsite) {
+            this.$('.o_anchors_loading').removeClass('d-none');
+            $.get(window.location.origin + urlInputValue, function (response) {
+                $selectMenu.empty();
+                _.each($(response).find(':not(p).oe_structure, :not(p)[data-oe-type=html]').children(), function (el) {
+                    if (el.id.length > 0) {
+                        anchorsArray.push('#' + el.id);
+                    }
+                });
+                _.map(anchorsArray, function (elem) {
+                    $selectMenu.append($('<option>', {text: elem}));
```

Comment:
Why not pushing the option in selectMenu directly instead of filling a table first then going through it to create the options ? :)

Reply:
That is what I did initially but whitespace, #top and #bottom (from anchorsArray) doesn't show up that's why I decided to fill selectMenu with _.map just after. Do you have an idea to fix that ?

Reply:
Can be done now